src/ecmascript_simd.js

```diff
@@ -538,18 +515,40 @@ if (typeof SIMD.int32x4 === "undefined") {
     get: function() { return tobool(this.w); }
   });
 
+}
+
+if (typeof SIMD.int32x4.allTrue === "undefined") {
+  /**
+    * Check if all 4 lanes hold a true value (bit 31 == 1)
+    * @param {int32x4} v An instance of int32x4.
+    * @return {Boolean} All 4 lanes holds a true value
+    */
+  SIMD.int32x4.allTrue = function(v) {
+    if (!(v instanceof SIMD.int32x4)) {
+      throw new TypeError("argument is not a int32x4.");
+    }
+    return ((v.x & 0x80000000) !== 0) &&
+           ((v.y & 0x80000000) !== 0) &&
+           ((v.z & 0x80000000) !== 0) &&
+           ((v.w & 0x80000000) !== 0);
```

Comment:
When is it possible to have 0x80000000 stored in an int32x4 lane? The int32x4 constructor does |0 on the values, so I thought we'd always be covered. I think it's desirable to avoid storing 0x80000000 in int32x4 SIMD lanes in the polyfill, since it's outside the signed int32 range.